src/traces/surface/convert.js

```diff
@@ -333,18 +378,17 @@ proto.refineCoords = function(coords) {
     var padWidth = 1 + width + 1;
     var padHeight = 1 + height + 1;
     var padImg = ndarray(new Float32Array(padWidth * padHeight), [padWidth, padHeight]);
+    var X = invert([], [
+        scaleW, 0, 0,
+        0, scaleH, 0,
+        0, 0, 1
+    ]);
```

Comment:
http://mathonline.wikidot.com/inverse-of-diagonal-matrices seems like just `1/scaleW` and `1/scaleH` then we can 🔪 `invert` too?

Reply:
Good call! I also removed some loops in https://github.com/plotly/plotly.js/pull/4930/commits/db6f9e7efe643bdab2df3fc168296e5c71158b1b.